src/screens/index.ts

```diff
@@ -1,13 +1,17 @@
 import { Navigation } from 'react-native-navigation';
-
 import Home from './home';
 import Counter from './counter';
 
-export const HOME = 'ueno-rns.Home';
-export const COUNTER = 'ueno-rns.Counter';
+export interface IScreen {
```

Comment:
I would move that into a typing file, if you end up adding interfaces, I find it easier to maintain it in one place dedicated to typedef

Reply:
It is a common practice to write and export interfaces in `.ts(x)` source files. Would you want to always write types in specific .d.ts files and them import them?

IMO Typing definition files (.d.ts) should only be used to polyfill packages that don't have typing definitions or to add a typed interface to untyped `.js` files included in the source.

Reply:
Well you are probably agree on that. I just see one case, if you have an interface you want to share between multiple components, where do you write the shared interface for these differents components?

Reply:
Then you would want to import them from where they should be living and relative to other stuff.

The classic scenario, for example, if you are building a Modal component, you would export the types with the component. 

```tsx
import { Modal, IModalOptions } from '~components/modal';

const options: IModalOptions = { visible: true };
Modal.create(options);
```

But if you have an interface that should be shared between components, like for example a form or something:

```tsx
import { Form } from '~components/form/Form';
import { Persist } from '~components/form/Persist';
import { IFormData } from '~components/form/interface';

const data: IFormData = { hello: 'world' };
const form = new Form(data);

return Persist<IFormData>(form);
```

or if you have a index, you can export them all together

```tsx
import { Form, Persist, IFormData } from '~components/form';

const data: IFormData = { hello: 'world' };
const form = new Form(data);

return Persist<IFormData>(form);
```


The reason I renamed and moved `IReactNativeNavigation` to `screens/IScreen` is mainly because the naming is not clear and just didn't know what to do with it.

And actually we shouldn't be using shared IProps for screens, they should be defined in screen component, and should rather use shared interfaces as properties, like:

```tsx
interface IProps {
  componentId: RNNComponentId,
}
```

or they may also be merged with an IScreen or what ever we end up calling it.